Update unreleased changelog with insecure http option (#275)
IronCoreLabs/ironcore-alloy: CHANGELOG.md

## Unreleased

### Breaking Changes

> - All SaaSShield configs now require an `allow_insecure_http` setting, which defaults to `false` in languages with defaults. This requires that the connection to the TSP is via HTTPS.

Reviewer: ```suggestion
- All SaaSShield configs now require an `allow_insecure_http` setting, which specifies whether or not the connection to the TSP must be via HTTPS. This defaults to `false` in languages with defaults.
```
